scanner/README.md

```diff
@@ -1,44 +1,102 @@
 
-Scanner requires the Go version be aligned with the [EXPECTED_GO_VERSION](../EXPECTED_GO_VERSION).
-This is verified when using any of Scanner's `make` targets.
+For local development, you can overwrite this restriction by specifying `EXPECTED_GO_VERSION` in the make targets that will depend on go tools, for example:
 
-### Local
+```
+make build EXPECTED_GO_VERSION=$(go version | { read _ _ v _; echo $v; })
+```
+
+### Running locally
+
+To run Scanner locally for development, copy the sample config and edit it to your liking:
+
+```sh
+cp config.sample.yaml config.yaml
+```
+
+Build Scanner and generate the development TLS certificates:
 
-Copy the sample config and edit it to your liking:
 ```sh
-$ cp config.sample.yaml config.yaml
+make build certs
 ```
 
-Generate the development TLS certificates:
+Run:
+
 ```sh
-$ make certs
+./bin/scanner -conf config.yaml
 ```
 
-Now you can build the dependencies and run scanner:
+The build system by default builds for `GOOS=linux`.  If you are running a non-Linux OS specify the `GOOS` yourself, or use `HOST_OS`.
+
 ```sh
-$ make deps
-$ go run cmd/scanner -conf config.yaml
+make GOOS='$(HOST_OS)' build
+```
+
+### Running standalone in Kubernetes
+
+Scanner contains a testing helm chart to deploy it standalone.  This is used for E2E testing or development.
+
+```
+make e2e-deploy
 ```
 
 ## scannerctl
 
-### Local
+There is a CLI that allows you to interact with Scanner, called [`scannerctl`](cmd/scannerctl/main.go).
+
+To build it, use:
+
+```sh
+make build
+```
+
+Or, specifically:
+
+```
+make bin/scannerctl
+```
+
+### Examples
 
-If you're connecting to your local scanner, you should already have the TLS certificates generated. If not, refer to
-running scanner in the above section.
+There are options to control how to run `scannerctl`.  See `scannerctl help`.
+
+#### Example 1: Connecting to local Scanner 
+
+A common use case is testing Scanner locally.  Once you have the local scanner build, certificates, and Scanner running with those certificates, you can run `scannerctl`:
 
-Once you have the certificates and scanner running with those certificates, you can run scannerctl:
 ```sh
-$ go run cmd/scannerctl/main.go \
-    -certs certs/client \
-    -server-name localhost \
-    -port 8443 \
+./bin/scannerctl scan \
+    --certs certs/scannerctl \
     https://registry.hub.docker.com/library/hello-world:latest
-```
\ No newline at end of file
+```
+
+
+
+#### Example 2: Connect to local Scanner in different modes
+
+Setup Scanner to run locally in different modes: 
+
+```sh
+sed 's@certs_dir: ""@certs_dir: certs/scanner-v4@' config.yaml.sample > matcher-config.yaml
+sed 's@certs_dir: ""@certs_dir: certs/scanner-v4@' config.yaml.sample > indexer-config.yaml
+sed -i '/matcher:/!b;n; s/enable: .*/enable: false/' indexer-config.yaml
+sed -i '/indexer:/!b;n; s/enable: .*/enable: false/' matcher-config.yaml
```

Comment:
> just a heads up, these don't work OOTB for macOS:

I don't think it's macOS, but probably your `sed` version? 

> also had an issue because the gRPC port was already in use. Maybe just post the example config file you used and make it clear it's possible users may need to adjust things?

4f907a8d

Reply:
Can we resolve this?

Reply:
What exactly it doesn't support, is it the `-i` option? So, `sed -i''` didn't work in this case?